kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala

```diff
@@ -106,23 +106,13 @@ class KyuubiSyncThriftClient private (
   }
 
   private def withRetryingRequest[T](block: => T, request: String): T = withLockAcquired {
-    var attemptCount = 1
-
-    var resp: T = null.asInstanceOf[T]
-    while (attemptCount <= maxAttempts && resp == null) {
-      try {
-        resp = block
-        remoteEngineBroken = false
-      } catch {
-        case e: TException if attemptCount < maxAttempts && isConnectionValid() =>
-          warn(s"Failed to execute $request after $attemptCount/$maxAttempts times, retrying", e)
-          attemptCount += 1
-          Thread.sleep(100)
-        case e: Throwable =>
-          error(s"Failed to execute $request after $attemptCount/$maxAttempts times, aborting", e)
-          throw e
-      }
-    }
+    val (resp, remoteEngineBrokenRetValue) = KyuubiSyncThriftClient.withRetryingRequestNoLock(
+      block,
+      request,
+      maxAttempts,
+      remoteEngineBroken,
+      isConnectionValid)
+    remoteEngineBroken = remoteEngineBrokenRetValue;
```

Comment:
The semantics seems a little different with previous. Before we only reset `remoteEngineBroken` if we succeeded to do request, and if failed the `remoteEngineBroken` depend on the probe so we do not change it.

So, we do not pass `remoteEngineBroken` to the new method `withRetryingRequestNoLock`, and change the semantics of the returned value `remoteEngineBrokenRetValue` to `shouldResetRemoteEngineBroken`, then if return true, we change  `remoteEngineBroken` to false, else we do nothing which is same with previous.

Reply:
@ulysses-you can you please re-check